### PR TITLE
Use GHC to compile C files

### PR DESCRIPTION
**Pull request by izgzhen**
https://github.com/snowleopard/hadrian/issues/349

### CODE REVIEW

**Comment by izgzhen** [comment]
The CI result is a bit funny -- Linux failed, OS X and Windows are fine.

**Comment by snowleopard** [comment]
Yep, looks like something platform-specific:
```
$ inplace/bin/ghc-stage2 -e 1+2
ghc-stage2: /home/travis/build/snowleopard/hadrian/ghc/_build/stage1/libraries/base/HSbase-4.10.0.0.o: unknown symbol `stat'
ghc-stage2: unable to load package `base-4.10.0.0'
```

In general, the PR looks fine, but I'd rename `GHCCompileC` to `CompileCWithGhc`.

**Comment by izgzhen** [comment]
Hmm, reproduced on my own Linux box ... will try to fix later

File: src/Settings/Builders/Ghc.hs [hunk 1]
```diff
@@ -15,6 +19,23 @@ ghcBuilderArgs = (builder (Ghc CompileHs) ||^ builder (Ghc LinkHs)) ? do
...
 
+ghcCbuilderArgs :: Args
+ghcCbuilderArgs =
+  builder (Ghc GHCCompileC) ? do
+    confCc  <- getSettingList . ConfCcArgs =<< getStage
+    confCpp <- getSettingList . ConfCppArgs =<< getStage
+    cppArgs <- getPkgDataList CppArgs
+    mconcat [ arg "-Wall"
+            , includeGhcArgs
+            , packageGhcArgs
```
Comment (bgamari) [inline src/Settings/Builders/Ghc.hs]: Shouldn't this include `commonGhcArgs` as well? On my machine I noticed that this patch resulted in linker errors apparently due to missing `-fPIC` flags during the RTS build. Presumably this is the reason.

Comment (izgzhen) [inline src/Settings/Builders/Ghc.hs]: Fixed, thanks

**Comment by snowleopard** [comment]
Looks good, merged!

**Pull request merged**